This patch fixes a regression where the device symlink to the pci address is
not created. Offending commit 6ae5ce8e8d4de666f31286808d2285aa6a50fa40,
cciss: remove redundant code.

This patch obsoletes the one I sent yesterday. It fixes a broken symlink in
sysfs. The previous patch appeared to work but was wrong as reported by Alan
Brunelle.
Please consider this for inclusion.

Signed-off-by: Mike Miller <mike.miller@hp.com>

diff --git a/drivers/block/cciss.c b/drivers/block/cciss.c
index 1e1f915..0f367b1 100644
--- a/drivers/block/cciss.c
+++ b/drivers/block/cciss.c
@@ -1365,6 +1365,7 @@ static void cciss_add_disk(ctlr_info_t *h, struct gendisk *disk,
 	disk->first_minor = drv_index << NWD_SHIFT;
 	disk->fops = &cciss_fops;
 	disk->private_data = &h->drv[drv_index];
+	disk->driverfs_dev = &h->pdev->dev;
 
 	/* Set up queue information */
 	blk_queue_bounce_limit(disk->queue, h->pdev->dma_mask);
